### PR TITLE
Приоритет операторов вместо рекурсивного спуска, другие исправления и оптимизации

### PR DESCRIPTION
**Pull request by Mr-Rm**
## Summary by CodeRabbit

* **Refactor**
  * Unified precedence-driven expression parser and simplified AST construction for more consistent parsing and node insertion.
  * Added clearer handling for unary vs. binary operator precedence.

* **Bug Fixes**
  * Improved handling of parentheses, calls, delimiters and ternary/question-operator edge cases.

* **Performance**
  * Reduced small-array allocations across node implementations to lower memory churn.

* **Tests**
  * Added parser tests covering operator and method delimiter scenarios.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by Mr-Rm** [comment]
Был баг в условном операторе. В качестве разделителей вместо запятых допускались и другие символы:
```bsl
Ф = ?(Истина? 1 ; 2);
```
В v1 этого бага нет

File: src/OneScript.Language/SyntaxAnalysis/DefaultBslParser.cs [hunk 1]
```diff
@@ -1233,134 +1209,68 @@ private BslSyntaxNode BuildExpression(NonTerminalNode parent, Token stopToken)
...
-            while (_lastExtractedLexem.Token == Token.And)
+            var firstArg = BuildPrimaryExpression();
+            while (LanguageDef.GetBinaryPriority(_lastExtractedLexem.Token) > prio)
             {
                 var operationLexem = _lastExtractedLexem;
                 NextLexem();
-                var secondArg = BuildNotExpression();
-                firstArg = MakeBinaryOperationNode(firstArg, secondArg, operationLexem);
-            }
+                var secondArg = BuildExpression(LanguageDef.GetBinaryPriority(operationLexem.Token));
```
Comment (EvilBeaver) [inline src/OneScript.Language/SyntaxAnalysis/DefaultBslParser.cs:1233]: Вот тут не надо ли проверить, что извлечен вообще оператор в токене, а не что-то не ожидаемое?
Иначе потом провалимся в BuildExpression(-1)->BuildPrimaryExpression() и там на строке 1250 пройдем дальше. Затем в потоке может оказаться какой-то валидный оператор и на строке 1264 ошибку потеряем. Не?

Comment (Mr-Rm) [reply src/OneScript.Language/SyntaxAnalysis/DefaultBslParser.cs:1233]: Нет, недопустимый токен с приоритетом -1 не пройдет через `while` в строке 1223.
Разве что попробовать уточнить диагностику

**Comment by EvilBeaver** [comment]
@cursor review

**Comment by Mr-Rm** [comment]
_Пока Cursor молчит_
Вот **это** - работает в 1С
```bsl
Асинх Функция Асинх(Асинх)
	Сообщить("Асинх=" + Асинх); 
	Асинх = Асинх + 1;
	Если Асинх <= 1 Тогда
		Возврат Асинх(Асинх + 1);
	КонецЕсли; 
	Возврат Асинх;
КонецФункции
; // обязательно оператор до вызова
Асинх(0)
```
И если функцию назвать `Ждать` - тоже, даже без предшествующего оператора

**Comment by EvilBeaver** [comment]
> _Пока Cursor молчит_ 

Он что-то перестал ревью делать, ага

> Вот **это** - работает в 1С

Асинх и его поведения вообще мало изучены и в 1скрипт сделано прям по наитию скорее, чем по принципу "как в 1С"
